File: Data Structures & Algorithms/decode-ways/submission-7.py
class Solution:
    def numDecodings(self, s: str) -> int:
        N = len(s)
        def dfs(i):
            if i ==N:
                return 1
            if s[i] == "0":
                return 0
            return dfs(i+1) + (dfs(i+2) if (i+1<N and 0<int(s[i:i+2])<=26)else 0)
        return dfs(0)

File: Data Structures & Algorithms/max-area-of-island/submission-7.java
class Solution {
    int ROWS,COLS;
    public int maxAreaOfIsland(int[][] grid) {
        ROWS = grid.length;
        COLS = grid[0].length;
        int res = 0;
        for(int i=0;i<ROWS;i++){
            for(int j=0;j<COLS;j++){
                res = Math.max(res,dfs(i,j,grid));
            }
        }
        return res;

        
    }

    public int dfs(int i,int j, int[][] grid) {
        if(i<0 || j<0 || i == ROWS || j == COLS || grid[i][j] != 1){
            return 0;
        }
        int res = 1;
        grid[i][j] = 0;
        res += dfs(i+1,j,grid);
        res += dfs(i-1,j,grid);
        res += dfs(i,j+1,grid);
        res += dfs(i,j-1,grid);
        return res;
    }
}
